<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>microTaboo by MohammedAlJaff</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body onkeypress="harambe(event)">
    <section class="page-header">
      <img id="logotaboo" src="Unknown.jpg" >
      <h1 class="project-name">microTaboo</h1>
      <h2 class="project-tagline">A general k-disjoint solver for Bioinformatic pipelines. </h2>
      <a href="https://github.com/MohammedAlJaff/microTaboo" class="btn">View on GitHub</a>
      <a href="https://github.com/MohammedAlJaff/microTaboo/zipball/Branch1" class="btn">Download .zip</a>
      <a href="https://github.com/MohammedAlJaff/microTaboo/tarball/Branch1" class="btn">Download .tar.gz</a>
      <a href="https://github.com/MohammedAlJaff/microTaboo/Branch1/zipball/ExampleRuns.zip" class="btn">ExampleRuns</a>
    </section>

    <section class="main-content">
      
</p>



<hr>

<h3>
<a id="overview" class="anchor" href="#overview" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>
<a id="overview" href="#overview"></a>Overview</h3>
<p>

A common challenge in bioinformatics is to identify unique sequences of a given size, where uniqueness is defined as the absence of other sequences that differ by less than a number of mismatches. Applications include finding sequences unique to a pathogen for infection diagnostics; identifying unique CRISPR target candidate regions and detecting phage or viral insertions. 
microTaboo is a tool that addresses these challenges and allows for efficient and extensive sequence mining of entire genomes for sequences up to 100 nucleotides in size in nucleotide space. 
In addition, microTaboo can also be used to find sequences in a FASTA file that only occur once. Moreover, if you want to find unique sequences that not only occur once in your FASTA file, but differ from any other sequence by a number of mismatches, then microTaboo should be your tool of choice.
<br><br>
microTaboo is able to run on Windows, Mac OS and Linux, running Java 7 or higher. Additionally, for smaller tasks (say FASTA files under 10mb), microTaboo can be run on a laptops/desktop computers with 8 or more gigabytes of RAM. 
</p>
<hr>

<h3>
<a id="key-concepts" class="anchor" href="#key-concepts" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>
<a id="dinmamma" href="#dinmamma"></a>Key concepts</h3>
<p><em><strong>Definition: The k-disjoint problem :</strong></em> 
<br><br>
(set-theoretic formulation)
<br><br>
Given two sets A and B containing sequences of length W, find all sequences X in A such that X is more than k mismatches away from any sequence Y in B. 
The result one obtains is the so-called <em><strong>k-disjoint</strong></em> set of A and B. 
The complement of the result is the <em><strong>k-intersection</strong></em> of A and B,
i.e the set of sequences in A that are at most k mismatches away from something in B.
<br><br>
alternativly
<br><br>
Let A and B be two sets containing sequences of length W. 
The k-disjoint of A and B is the set containing all sequences X in A such that X is at least k edit distances away from Y for all Y in B. 
Finding the k-disjoint of A and B is called the k-disjoint problem.
("string & text" formulation)
<br><br>
Given two fixed positive integer value parameters W and k aswell as two strings P and T where P has length n and T has length m, 
the k-disjoint problem consists of finding all positions i (0<i<n+1) which have the property that a W length substring of P 
beginning at i is at least k edit distances away from any W long substring of T.
<hr>
</p>
<h3>
<a id="how-to-use-microtaboo-to-solve-you-k-mismatch-problem" class="anchor" href="#how-to-use-microtaboo-to-solve-you-k-mismatch-problem" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>
<a id="getting-started" href="#getting-started"></a>
How to use microTaboo to solve you k-mismatch problem</h3>
<br><br>
<font color="red">Note: microTaboo requires Java SE Runtime Environment 7</font> <br><br>
<p>Below we give a step-by-step procedure of how to use microTaboo:<br><br>

1. Create a folder/directory, you can name this folder/directory whatever you want as long as you know where it is. For the sake of example, we call our folder/directory F.<br><br>

2. Download the microTaboo.jar file from the website and place it in the folder F.<br><br>

3. Create three new (sub) folders/directories inside our main folder F. For simplicity and consistency, we call these new folders A, B and R. <br><br>

4. Inside A, we should now put the FASTA files that we want our sequences to be found in. Inside the folder B we should place all FASTA files we do not want all sequences not to be inside in. As an added benefit microTaboo is able to deal with subfolders inside the folder B so if you would like to group your FASTA files, you can.

5. To run microTaboo, open a command line tool and navigate to the folder F and type the below formula:<br><br>

&gt; java -Xmx[a number]m -jar microTaboo.jar [name of folder A] [name of folder B] [name of folder R] [sequence length W] [number of mismatches k] d/i/a [number of cores] <br><br>

For our example, we would write the following:<br>
&gt;  java -Xmx6000m -jar microTaboo.jar A B R 20 3 d 3;<br>
That is, we're specifying that we would like  microTaboo to give us the k-disjoint, where k=3, of the contents in A and B  and put the result files into the subfolder R.  
In other words, we would like microTaboo to give us a file containing all sequences of length 20 in A that are at least 3 mismatches away from anything/any sequence in B.
<br>
There you go. You’ve hopefully now run microTaboo successfully.
<br>
A zip file called ExampleRuns.zip is provided on the github page. Unzipping the file gives you a folder containing three subfolders
where each subfolder contains a test case with a readme files. These are meant to be downloaded and used as a verification to see
if microTaboo actually runs on your system. Follow the instructions in the readme files, not only do you check if everything runs
on your machine, it also gives you some practice on microTaboo.
</p>

<hr>

<h3>
<a id="how-to-use-microtaboo-to-find-sequences-in-a-fasta-file-that-occur-only-once" class="anchor" href="#how-to-use-microtaboo-to-find-sequences-in-a-fasta-file-that-occur-only-once" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>
<a id="getting-started" href="#getting-started"></a>
How to use microTaboo to find sequences in a FASTA file that occur only once</h3>

<p>In this section, we'll show how microTaboo can be used to find all sequences of 
length in a FASTA file, eg a whole genome sequence FASTA, that occur only once. In other words, here we'll try to 
find only unique sequences in a FASTA file relative to it self. Below are instructions on how you could
do this with microTaboo.<br></br>

1. Create two separate folders (A and B), each one containing the FASTA file(s) for the same organism of interest. 
Also create a third folder (R) where the result should be saved. We call these folders A, B and R for simplicity.
<br><br>
2. Run microTaboo from the command line by typing: <br>
java -Xmx[#]m -jar path/microTaboo.jar path/A path/B path/R [#w] [#k] [d] [#cores] [m] <br><br>
-Xmx[#]m determines the maximum amount of memory (in MBs) microTaboo can use. <br>
The path to microTaboo.jar, A, B and R can be either relative or absolute.  <br>
#w stands for word length (in bases) and should be a multiple of 5 and atleast 10 (i.e 10, 15, 20 etc).<br>
#k is the mismatch threshold, can be any number greater than or equal to 0. It is recommended to not use #k greater than 5.<br>
d means disjoint and which corresponds to unique sequences. <br>
#cores decides how many cores microTaboo should use, minimum is 1.<br>
m is a flag which is necessary when running a FASTA file against itself. <br><br>

Example:<br>
java -Xmx6000m -jar path/microTaboo.jar path/A path/B path/R 20 2 d 5 m <br>
Means finding all unique sequnces within 2 mismatches of length 20 bases while using 5 cores and allowing microTaboo to utilize a maximum of 6GB of RAM.

<br><br><br>

<font color="red">Note on the presence of uncertain bases in a genome/FASTA-file.</font>
<p>
When taking into account of unknown/uncertain bases (N), microTaboo follows the self-imposed convention that N compared to N gives 1 mismatch. 
Because of this, sequences containing many unknown bases, ie many N’s, will have a high mismatch count when compared to any other sequence.
In the extreme case where we have a W nt sequence of only Ns; when compared to itself, the number of mismatches will be W. 
If a microTaboo run is initiated with a mismatch parameter k < W, these sequences will be included in either the disjoint set or unique set.
<br><br>
Additionally, should unknown bases other than N's be present in an input FASTA file, microTaboo will convert them to N's and
treat them as such.
</p>
<br><br>

</p>

</code></pre>


      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/MohammedAlJaff/microTaboo">microTaboo</a> is maintained by 
        <a href="https://github.com/MohammedAlJaff">Mohammed Al-Jaff</a> & <a href="https://github.com/eriqual">eriqual</a>..</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

  
  </body>
</html>
