<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>microTaboo by MohammedAlJaff</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">microTaboo</h1>
      <h2 class="project-tagline">A general k-mismatch solver for Bioinformatic pipelines. </h2>
      <a href="https://github.com/MohammedAlJaff/microTaboo" class="btn">View on GitHub</a>
      <a href="https://github.com/MohammedAlJaff/microTaboo/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/MohammedAlJaff/microTaboo/tarball/master" class="btn">Download .tar.gz</a>
    </section>

    <section class="main-content">
      <p>&lt;!DOCTYPE html&gt;
</p>

<p></p>
    
    
    

<pre><code>&lt;link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css"&gt;

&lt;title&gt;microTaboo&lt;/title&gt;
</code></pre>

<p></p>

<p></p>

<pre><code>&lt;!-- HEADER --&gt;
&lt;div id="header_wrap" class="outer"&gt;
    &lt;header class="inner"&gt;
      &lt;a id="forkme_banner" href="https://github.com/MohammedAlJaff/microTaboo"&gt;View on GitHub&lt;/a&gt;

      &lt;h1 id="project_title"&gt;microTaboo&lt;/h1&gt;
      &lt;h2 id="project_tagline"&gt;a RAM-efficient tool for drywejfoiwefoiwhfuwufh cleaning&lt;/h2&gt;

        &lt;section id="downloads"&gt;
          &lt;a class="zip_download_link" href="https://github.com/nedaz/kraken/zipball/master"&gt;Download this project as a .zip file&lt;/a&gt;
          &lt;a class="tar_download_link" href="https://github.com/nedaz/kraken/tarball/master"&gt;Download this project as a tar.gz file&lt;/a&gt;
        &lt;/section&gt;
    &lt;/header&gt;
&lt;/div&gt;

&lt;!-- MAIN CONTENT --&gt;
&lt;div id="main_content_wrap" class="outer"&gt;
  &lt;section id="main_content" class="inner"&gt;
    &lt;h3&gt;
</code></pre>

<p><a id="methods" href="#methods"></a>Article</p>

<p><a href="http://www.biomedcentral.com/1471-2105/15/227?utm_source=dlvr.it&amp;utm_medium=tumblr">Place link here</a></p>

<p>citation here</p>

<hr>

<h3>
<a id="overview" class="anchor" href="#overview" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>
<a id="overview" href="#overview"></a>Overview</h3>

<p> Something something k-mismatch problem. 
Mention also that microTaboo can also be  used to find sequences in a fasta file that only occur once. 
Moreover if you want to find unique sequences that are not only occur once in your fasta file, but differ from any other sequence
by a number of mismatches, then microTaboo should be your app of choice.</p>

<hr>

<h3>
<a id="key-concepts" class="anchor" href="#key-concepts" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>
<a id="dinmamma" href="#dinmamma"></a>Key concepts</h3>
<p><em><strong>k-mismatch problem:</strong></em> Given two sets A and B containing sequences of length W, find all sequences X in A such that X is more than k mismatches away from any sequence Y in B. The result one obtains is the so-called <em><strong>k-disjoint</strong></em> set of A and B. The complement of the result is the <em><strong>k-intersection</strong></em> of A and B, i.e the set of sequences in A that are at most k mismatches away from something in B.</p>

<hr>

<h3>
<a id="how-to-use-microtaboo-to-solve-you-k-mismatch-problem" class="anchor" href="#how-to-use-microtaboo-to-solve-you-k-mismatch-problem" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>
<a id="getting-started" href="#getting-started"></a>
How to use microTaboo to solve you k-mismatch problem</h3>

<p>Below we give a step-by-step procedure of how to use microTaboo:<br><br>

1. Create a folder/directory, you can name this folder/directory whatever you want as long as you know where it is. For the sake of example, we call our folder/directory F.<br><br>

2. Download the microTaboo.jar file from the website and place it in the folder F.<br><br>

3. Create three new (sub) folders/directories inside our main folder F. For simplicity and consistency, we call these new folders A, B and R. <br><br>

4. Inside A, we should now put the FASTA files that we want our sequences to be found in. Inside the folder B we should place all FASTA files we do not want all sequences not to be inside in. As an added benefit microTaboo is able to deal with subfolders inside the folder B so if you would like to group your FASTA files, you can.<br><br>

5. To run microTaboo, open a command line tool and navigate to the folder F and type the below formula:<br><br>

&gt; java -Xmx[a number]m -jar microTaboo.jar [name of folder A] [name of folder B] [name of folder R] [sequence length W] [number of mismatches k] d/i/a [number of cores] <br><br>

For our example, we would write the following:<br>
&gt;  java -Xmx6000m -jar microTaboo.jar A B R 3 d 3;<br>
That is, we're specifying that we would like  microTaboo to give us the k-disjoint, where k=3, of the contents in A and B  and put the result files into the subfolder R.  In other words, we would like microTaboo to give us a file containing all sequences of length W in A that are at least 3 mismatches away from anything/any sequence in B.
<br>
There you go. Youâ€™ve hopefully now run microTaboo successfully.
<br>
A zip file called ExampleRuns.zip is provided on the github page. Unzipping the file gives you a folder containing three subfolders
where each subfolder contains a test case with a readme files. These are meant to be downloaded and used as a verification to see
if microTaboo actually runs on your system. Follow the instructions in the readme files, not only do you check if everything runs
on your machine, it also gives you some practice on microTaboo.
</p>

<hr>

<h3>
<a id="how-to-use-microtaboo-to-find-sequences-in-a-fasta-file-that-occur-only-once" class="anchor" href="#how-to-use-microtaboo-to-find-sequences-in-a-fasta-file-that-occur-only-once" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>
<a id="getting-started" href="#getting-started"></a>
How to use microTaboo to find sequences in a FASTA file that occur only once</h3>

<p>In this section, we'll show how microTaboo can be used to find all sequences of 
length in a FASTA file, eg a whole genome sequence FASTA, that occur only once. In other words, here we'll try to 
find only unique sequences in a FASTA file relative to its self. Below is a step by step instruction on how you could
do this with microTaboo. 



:<br><br>

</p>

<pre><code>  &lt;/section&gt;
&lt;/div&gt;

&lt;!-- FOOTER  --&gt;
&lt;div id="footer_wrap" class="outer"&gt;
  &lt;footer class="inner"&gt;
    &lt;p class="copyright"&gt;microTaboo "maintained" by &lt;a href="https://github.com/MohammedAlJaff"&gt;papa smurf&lt;/a&gt; &amp; 
    &lt;a href="https://github.com/eriqual"&gt;Ghostface Killah&lt;/a&gt;&lt;/p&gt;
    &lt;p&gt;Published with &lt;a href="http://pages.github.com"&gt;GitHub Pages&lt;/a&gt;&lt;/p&gt;
  &lt;/footer&gt;
&lt;/div&gt;
</code></pre>

<p>
</p>

      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/MohammedAlJaff/microTaboo">microTaboo</a> is maintained by <a href="https://github.com/MohammedAlJaff">MohammedAlJaff</a>.</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

  
  </body>
</html>
